packages/contract-wrappers/test/coordinator_wrapper_test.ts

```diff
@@ -0,0 +1,359 @@
+                    // Not used here when instantiating with the imported app
+                    RPC_URL: 'http://ignore',
+                },
+            },
+            // Optional selective delay on fill requests
+            SELECTIVE_DELAY_MS: 0,
+            EXPIRATION_DURATION_SECONDS: 60, // 1 minute
+        };
+        app = await getAppAsync(
+            {
+                [config.networkId]: provider,
+            },
+            coordinatorServerConfigs,
+        );
+
+        await app.listen(coordinatorServerConfigs.HTTP_PORT, () => {
+            // tslint:disable-line:await-promise
+            console.log(`Coordinator SERVER API (HTTP) listening on port ${coordinatorServerConfigs.HTTP_PORT}!`); // tslint:disable-line:no-console
+        });
+
+        // setup coordinator registry
+        coordinatorRegistryInstance = new CoordinatorRegistryContract(
+            CoordinatorRegistry.compilerOutput.abi,
+            contractAddresses.coordinatorRegistry,
+            provider,
+            { gas: 6000000 },
+        );
+
+        await web3Wrapper.awaitTransactionSuccessAsync(
+            await coordinatorRegistryInstance.setCoordinatorEndpoint.sendTransactionAsync(coordinatorEndpoint, {
+                from: feeRecipientAddress,
+            }),
+            constants.AWAIT_TRANSACTION_MINED_MS,
+        );
+    });
+    after(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+        signedOrder = await fillScenarios.createFillableSignedOrderWithFeesAsync(
+            makerAssetData,
+            takerAssetData,
+            new BigNumber(1),
+            new BigNumber(1),
+            makerAddress,
+            takerAddress,
+            fillableAmount,
+            feeRecipientAddress,
+        );
+        anotherSignedOrder = await fillScenarios.createFillableSignedOrderWithFeesAsync(
+            makerAssetData,
+            takerAssetData,
+            new BigNumber(1),
+            new BigNumber(1),
+            makerAddress,
+            takerAddress,
+            fillableAmount,
+            feeRecipientAddress,
+        );
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    describe('test setup', () => {
+        it('should have coordinator registry which returns an endpoint', async () => {
+            const recordedCoordinatorEndpoint = await coordinatorRegistryInstance.getCoordinatorEndpoint.callAsync(
+                feeRecipientAddress,
+            );
+            expect(recordedCoordinatorEndpoint).to.be.equal(coordinatorEndpoint);
+        });
+        it('should have coordinator server endpoint which responds to pings', async () => {
+            const result = await fetchAsync(`${coordinatorEndpoint}/v1/ping`);
+            expect(result.status).to.be.equal(200);
+            expect(await result.text()).to.be.equal('pong');
+        });
+    });
```

Comment:
Let's add some failure cases. Some that come to mind are:

- `feeRecipientAddress` is not in registry.
- `coordinatorEndpoint` in registry is incorrect. 

An additional edge-case:
- Batch fill with 2 orders each specifying a different coordinator server